docs/source/API/core/policies/RangePolicy.rst

```diff
@@ -97,27 +105,58 @@ Public Class Members
 Constructors
 ~~~~~~~~~~~~
 
+.. cppkokkos:function:: ChunkSize(int value_)
+
+   Provide a hint for optimal chunk-size to be used during scheduling.
+   For the SYCL backend, the workgroup size used in a ``parallel_for`` kernel can be set via this passed to ``RangePolicy``.
+
 .. cppkokkos:function:: RangePolicy()
 
    Default Constructor uninitialized policy.
 
-.. cppkokkos:function:: template<class ... InitArgs> RangePolicy(const int64_t& begin, const int64_t& end, const InitArgs ... init_args)
+.. cppkokkos:function:: RangePolicy(int64_t begin, int64_t end)
 
-   Provide a start and end index as well as optional arguments to control certain behavior (see below).
+   Provide a start and end index.
 
-.. cppkokkos:function:: template<class ... InitArgs> RangePolicy(const ExecutionSpace& space, const int64_t& begin, const int64_t& end, const InitArgs ... init_args)
+.. cppkokkos:function:: RangePolicy(int64_t begin, int64_t end, ChunkSize chunk_size)
 
-   Provide a start and end index and an ``ExecutionSpace`` instance to use as the execution resource, as well as optional arguments to control certain behavior (see below).
+   Provide a start and end index as well as a ``ChunkSize``.
 
-Optional ``InitArgs``:
-^^^^^^^^^^^^^^^^^^^^^^
+.. cppkokkos:function:: RangePolicy(const ExecutionSpace& space, int64_t begin, int64_t end)
 
-* ``ChunkSize`` : Provide a hint for optimal chunk-size to be used during scheduling. For the SYCL backend, the workgroup size used in a ``parallel_for`` kernel can be set via this variable.
+   Provide a start and end index and an ``ExecutionSpace`` instance to use as the execution resource.
+
+.. cppkokkos:function:: RangePolicy(const ExecutionSpace& space, int64_t begin, int64_t end, ChunkSize chunk_size)
+
+   Provide a start and end index and an ``ExecutionSpace`` instance to use as the execution resource, as well as a ``ChunkSize``.
 
 Preconditions:
+^^^^^^^^^^^^^^
 
 * The start index must not be greater than the end index.
```

Comment:
You referred to the PR that check that the indices arguments are representable as `index_type`.
Maybe that would be confusing because we pretend that the constructors take member_type/index_type by value when in reality they are templated.
Can we really pretend it is QOI?
(this does not need to be resolved here)

Reply:
I added a sentence under "Preconditions" (I might move it, though).